src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs

```diff
@@ -44,9 +44,9 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             {
                 // look for access token in the cache first.
                 // no access token is found, then it means token does not exist
-                // or new assertion has been passed. We should not use Refresh Token
-                // for the user because the new incoming token may have updated claims
-                // like MFA etc.
+                // or new assertion has been passed. 
+                // Look for a refresh token, if refresh token is found perform refresh token flow.
```

Comment:
so...how is the above comment getting addressed?

Reply:
Earlier we did not look for the refresh token for obo and instead if AT is not found would fetch a new AT. Now with the new [API review](https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?version=GBdev&path=%2FConfidentialClient%2FOBO.md&_a=preview) we look for a refresh token filtered by the user assertion in the request. If a refresh token is found we use it. If a RT is not found we fetch the access token.

Reply:
I will update the comment to make it more clear